### ISSUE TITLE
Doesn't work under KDE based distros

### ISSUE THREAD

**Issue by mbnoimi** [clearkimura/Customizer#36]
As shown in the log I'm unable to load any ISO image by Customizer:
P.S. I'm using Mint KDE 15

```
mbnoimi-pc mbnoimi # gksu /opt/Customizer/GUI.gambas

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking if GUI is locked
[EVENT] Locking GUI

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking configuration file
[EVENT] Checking essential directories and files
[EVENT] Disabling widgets
[EVENT] Checking configuration file

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: MOUNT_DIR=
[DEBUG] Def: /media/ISO
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: MESSAGES_COLORS=
[DEBUG] Def: 1
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: MESSAGES_COLORS=
[DEBUG] RpcStr: 1
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: FORCE_CHROOT=
[DEBUG] Def: 0
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: FORCE_CHROOT=
[DEBUG] RpcStr: 1
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: APT_HELPER=
[DEBUG] Def: 1
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: APT_HELPER=
[DEBUG] RpcStr: 1
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: BOOT_FILES=
[DEBUG] Def: 0
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: VRAM=
[DEBUG] Def: 256
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: VRAM=
[DEBUG] RpcStr: 256
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: RESOLUTION=
[DEBUG] Def: 800x600
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: RESOLUTION=
[DEBUG] RpcStr: 1024x768
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: COMPRESSION=
[DEBUG] Def: tar
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: COMPRESSION=
[DEBUG] RpcStr: xz
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: LOCALES=
[DEBUG] Def: 
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: LOCALES=
[DEBUG] RpcStr: C
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: ISO=
[DEBUG] Def: 
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: ISO=
[DEBUG] RpcStr: /media/truecrypt3/~Files/Programs/Linux/System/Mint/linuxmint-15-kde-dvd-64bit.iso
[DEBUG] Loading file: /opt/Customizer/settings.conf
[DEBUG] Saving file: /opt/Customizer/settings.conf
================ Replace_Str() end (found) ================

[EVENT] Extracting ISO
[EVENT] Detecting Terminal Emulator
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.
[EVENT] Checking essential directories and files
[EVENT] Disabling widgets
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus
[1]mbnoimi-pc mbnoimi # 
```

the content of /opt/Customizer/settings.conf:

```
# Main settings
WORK_DIR="/home/mbnoimi/Projects/ECSBUNTU/client/Customizer"
MOUNT_DIR="/home/mbnoimi/Projects/ECSBUNTU/client/Iso"

# Preferences
MESSAGES_COLORS="1"
FORCE_CHROOT="1"
APT_HELPER="1"
RESOLUTION="1024x768"
BOOT_FILES="0"
VRAM="256"
COMPRESSION="xz"
LOCALES="C"

# Saved variables
ISO="/media/truecrypt3/~Files/Programs/Linux/System/Mint/linuxmint-15-kde-dvd-64bit.iso"
DEB=""
HOOK=""
PIC=""
```

**Comment by ghost** [comment]
@mbnoimi 
Firstly, let me confirm this issue:
1. You are using Linux Mint 15 KDE host (? bit) system, and trying to customize ISO of same distro (64-bit). IMPORTANT: To customize a 64-bit ISO, make sure that you customize on 64-bit host system.
2. I see you run Customizer GUI from Terminal. But based on the log, you probably set the path to ISO first from Terminal, then only run the GUI, correct?
3. And most importantly, what did happen after you run the `gksu /opt/Customizer/GUI.gambas` command, visually speaking? Did GUI show up at all? Any screenshot that can explain this issue?

**Comment by mbnoimi** [comment]
@clearkimura 

> You are using Linux Mint 15 KDE host (? bit)

I'm using Mint KDE 15 x64 and I want to customize a new distro from it.

>  see you run Customizer GUI from Terminal. But based on the log, you probably set the path to ISO first from Terminal, then only run the GUI, correct?

Nop, I'm using GUI but I only run  gksu /opt/Customizer/GUI.gambas to see the log. Every I did done in GUI.

> And most importantly, what did happen after you run the gksu /opt/Customizer/GUI.gambas command, visually speaking? Did GUI show up at all? Any screenshot that can explain this issue?

I definitely saw the GUI but it I get disabled buttons because Customizer didn't load the ISO after selecting it!

1st step: 
![snapshot-1](https://f.cloud.github.com/assets/284547/1583551/e8ebc2ce-51fe-11e3-91fc-4149818d9f89.png)

2nd step: 
![snapshot-3](https://f.cloud.github.com/assets/284547/1583554/fd0fd16e-51fe-11e3-84d9-2d36219e668d.png)

P.S. If you want I can upload a snapshot of my desktop to see the exact procedures.

**Comment by ghost** [comment]
@mbnoimi 
Based on very few feedbacks, there is at least one user had success for customizing vanilla Ubuntu 13.04 64-bit image. But so far no feedback received for Linux Mint 64-bit ISO image file.

Anyway, try either one of below:
1. Select other Ubuntu-based distro 64-bit ISO image file ( for example, Ubuntu 13.04 64-bit ), and see if the ISO image file is extracted successfully?
2. Send a screenshot of Linux Mint 15 KDE ISO image folder structure ( Right-click on *.iso image file, select `Browse this folder` or similar option, a File manager will open and show few directories and files. Take this screenshot by `Alt`+`PrtScrn` and post this screenshot in your reply ).

And kindly feedback above result to us.

P.S.: Linux Mint 15 is based on Ubuntu 13.04 package base, that's why I suggest to try selecting Ubuntu 13.04 64-bit ISO image file

**Comment by mbnoimi** [comment]
@clearkimura 

> Select other Ubuntu-based distro 64-bit ISO image file ( for example, Ubuntu 13.04 64-bit ), and see if the ISO image file is extracted successfully?

I selected ubuntu-12.04.3-server-amd64.iso but unfortunately I get same behavior. 

> Send a screenshot of Linux Mint 15 KDE ISO image folder structure 

I didn't understand you clearly but the ISO mount directory is empty and below a screenshot of Mint KDE 15 x64
![snapshot-4](https://f.cloud.github.com/assets/284547/1592712/67edda5a-52c0-11e3-842f-33096ee18065.png)

**Comment by ghost** [comment]
@mbnoimi 
Commented for 1 and 2. Please try option 3 and 4 as explained below.
1. Ok, same behaviour happened.
2. Hm.. the directory structure of ISO doesn't seemed to be different than Ubuntu. Nevermind this.
3. [New] I noticed that your working directory is quite deep far away from /home. Instead of above, please try to use default working directory like this:

```
WORK_DIR="/home"
MOUNT_DIR="/mnt"
```

and select the same Linux Mint 15 KDE ISO file one more time to see if it works?
1. [New] I also noticed that your source ISO is stored in an encrypted (?) partition/directory. This might be the cause of this issue. If the ISO file is stored in an encrypted volume, no other application could read it ( except Truecrypt ). So, please try to store the ISO file somewhere unencrypted, for example your Home directory, like this:

```
ISO="/home/$USER/linuxmint-15-kde-dvd-64bit.iso"
```

and select the same Linux Mint 15 KDE ISO file one more time to see if it works?

P.S.: $USER is just a variable. For example, if your username is mbnoimi, then your ISO path should be:

```
ISO="/home/mbnoimi/linuxmint-15-kde-dvd-64bit.iso"
```

**Comment by mbnoimi** [comment]
> [New] I noticed that your working directory is quite deep far away from /home. Instead of above, please try to use default working directory 

see my new configs (/opt/Customizer/settings.conf):

```
WORK_DIR="/home"
MOUNT_DIR="/mnt"
MESSAGES_COLORS="1"
FORCE_CHROOT="0"
APT_HELPER="1"
RESOLUTION="1024x768"
BOOT_FILES="0"
VRAM="1024"
COMPRESSION="xz"
LOCALES="C"
ISO="/home/mbnoimi/linuxmint-15-kde-dvd-64bit.iso"
DEB=""
HOOK=""
PIC=""
```

> I also noticed that your source ISO is stored in an encrypted (?) partition/directory.... etc

After running Customizer as above configs I got the following (it hanged at QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus line)

```
bnoimi-pc mbnoimi # gksu /opt/Customizer/GUI.gambas

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking if GUI is locked
[EVENT] Locking GUI

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking configuration file
[EVENT] Checking essential directories and files
[EVENT] Disabling widgets

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: ISO=
[DEBUG] Def: 
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf                                                                                                                                     
[DEBUG] SrchStr: ISO=                                                                                                                                                               
[DEBUG] RpcStr: /home/mbnoimi/linuxmint-15-kde-dvd-64bit.iso                                                                                                                        
[DEBUG] Loading file: /opt/Customizer/settings.conf                                                                                                                                 
[DEBUG] Saving file: /opt/Customizer/settings.conf                                                                                                                                  
================ Replace_Str() end (found) ================                                                                                                                         

[EVENT] Extracting ISO                                                                                                                                                              
[EVENT] Detecting Terminal Emulator                                                                                                                                                 
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.                                                                               
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.                                                                               
[EVENT] Checking essential directories and files                                                                                                                                    
[EVENT] Disabling widgets
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus
^C
[130]mbnoimi-pc mbnoimi # ^C
[130]mbnoimi-pc mbnoimi #
```

**Comment by ghost** [comment]
@mbnoimi 
You are right. I have compared your log with mine ( loading Xubuntu 12.04 ISO image file ).

You log is showing some lines of errors with `Disabling widgets` message:

> [EVENT] Extracting ISO
> [EVENT] Detecting Terminal Emulator
> QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.
> QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.
> [EVENT] Checking essential directories and files
> [EVENT] Disabling widgets
> QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
> QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus

In contrast, a normally extracted ISO will show no errors and `Enabling widgets` message:

> [EVENT] Extracting ISO
> [EVENT] Detecting Terminal Emulator
> [EVENT] Checking essential directories and files
> [EVENT] Enabling widgets

So both of us suspects that some KDE components doesn't play well.
Earlier, you tried with a 64-bit Server ISO file. That didn't work for unclarified reason.

This time, please try again with other 64-bit Desktop ISO image file. Preferably select Ubuntu 12.04 ( or newer ), 64-bit Desktop ISO image file, with Unity, GNOME, XFCE or LXDE environment.
And feedback to us the result.

**Comment by ghost** [comment]
@mbnoimi 
To summarize previous comment:

Linux Mint 15 KDE might not be supported by Customizer. To confirm this, please try with other Ubuntu or Linux Mint (preferably non-KDE) ISO images. This is the last trial before we can reach a conclusion.

Kindly feedback to us the result.

**Comment by mbnoimi** [comment]
> Linux Mint 15 KDE might not be supported by Customizer. To confirm this, please try with other Ubuntu or Linux Mint (preferably non-KDE) ISO images. This is the last trial before we can reach a conclusion.

After downloading latest ubuntu image (ubuntu-13.10-desktop-amd64.iso) I got exact same bug as show below!!!

--- /opt/Customizer/settings.conf ---

```
WORK_DIR="/home"
MOUNT_DIR="/mnt"
MESSAGES_COLORS="1"
FORCE_CHROOT="0"
APT_HELPER="1"
RESOLUTION="1024x768"
BOOT_FILES="0"
VRAM="1024"
COMPRESSION="xz"
LOCALES="C"
ISO="/home/mbnoimi/ubuntu-13.10-desktop-amd64.iso"
DEB=""
HOOK=""
PIC=""
```

--- Terminal log ---

```
mbnoimi-pc mbnoimi # gksu /opt/Customizer/GUI.gambas

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking if GUI is locked
[EVENT] Locking GUI

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: WORK_DIR=
[DEBUG] Def: /home
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================

[EVENT] Checking configuration file
[EVENT] Checking essential directories and files
[EVENT] Disabling widgets

===================== Get_Str() begin =====================
[DEBUG] SrchFile: /opt/Customizer/settings.conf
[DEBUG] SrchStr: ISO=
[DEBUG] Def: 
[DEBUG] Loading file: /opt/Customizer/settings.conf
================== Get_Str() end (found) ==================


=================== Replace_Str() begin ===================
[DEBUG] SrchFile: /opt/Customizer/settings.conf                                                                                                                                     
[DEBUG] SrchStr: ISO=                                                                                                                                                               
[DEBUG] RpcStr: /home/mbnoimi/ubuntu-13.10-desktop-amd64.iso                                                                                                                        
[DEBUG] Loading file: /opt/Customizer/settings.conf                                                                                                                                 
[DEBUG] Saving file: /opt/Customizer/settings.conf                                                                                                                                  
================ Replace_Str() end (found) ================                                                                                                                         

[EVENT] Extracting ISO                                                                                                                                                              
[EVENT] Detecting Terminal Emulator                                                                                                                                                 
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.                                                                               
QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.                                                                               
[EVENT] Checking essential directories and files                                                                                                                                    
[EVENT] Disabling widgets
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus
[1]mbnoimi-pc mbnoimi # 
```

**Comment by ghost** [comment]
@mbnoimi 
Thank you for continuous feedback to us.

You are running Customizer on KDE host with KDE ISO image file.
I have been assuming the Qt components on ISO image side had compatibility issue.
I should suspect on KDE host side first..
My bad for misunderstanding.

I should have asked below questions first:
- Which branch of Customizer are you running: main or gambas3?
- How did you install Customizer: install.sh script or PPA? 
- Did you see any error when installing Customizer at first time?

Assuming that gambas3 is readily available in your repo;
Try to reinstall Customizer (gambas3) using install.sh script via Terminal:

```
 wget https://dl.dropboxusercontent.com/u/54183088/install.sh
 chmod +x install.sh
 sudo ./install.sh -b gambas3 -i
```

Then, select again the same ISO i.e. ubuntu-13.10-desktop-amd64.iso in the newly reinstalled one.

And feedback to us the result.

P.S.: This weekend, I might try to reproduce this issue on my side also. But I will be using vanilla Linux Mint 15 (non-KDE). We will confirm this issue together.

P.S.S.: My apologies, I gave up on trying on my side. My hardware is quite aged for heavyweight desktop environments (Unity, Cinnamon, KDE, GNOME and alike). Too slow even before installation. I am trying out other ways to check and waiting for your reply..

---

2013.11.30
My latest findings reveal that KDE Plasma Desktop components have this kind of bug. And this happens if you launch any application using incorrect command. I will explain this once I received your feedback.
We are very close to solving this issue.

**Comment by mbnoimi** [comment]
> Which branch of Customizer are you running: main or gambas3?

I'm using gambas3's branch

> How did you install Customizer: install.sh script or PPA? 

by install.sh

> Did you see any error when installing Customizer at first time?

No; Everything went well

> Try to reinstall Customizer (gambas3) using install.sh script via Terminal:

I got same exact bug as shown in the full log here: http://pastebin.com/JfU2QPN1

>  We are very close to solving this issue. 

Thanks very much for caring.

**Comment by ghost** [comment]
@mbnoimi 
First of all, I found that KDE host system was causing this issue.

> QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.
> ...
> QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
> QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus

This issue affects any applications, including KDE popular applications such as Kate, Ktorrent, etc. To be exact, running any GUI application from command line may experience this issue.

**Problem**
Using `sudo` or `gksu` to run any GUI application from command line in KDE

**Solution**
Use `kdesudo` instead (Older KDE may use `kdesu`)

I have added this as **IMPORTANT** remarks to [FAQ](https://github.com/clearkimura/Customizer/wiki/FAQ), [README.md](https://github.com/clearkimura/Customizer/blob/gambas3/README.md) and [User Guide](https://docs.google.com/document/d/1PfhHnSBjv-IDI7Yh5obhMGYCAV9Gw1NPEynU4GqKTsA/edit).

Hence below:
- Run Customizer GUI on KDE host system by:
  
  ```
    kdesudo /opt/Customizer/GUI.gambas
  ```
- If you still cannot run the GUI for unknown reason, reset your KDE settings by renaming or deleting the `~/.kde4` folder under your Home directory. KDE will newly create another `~/.kde4` folder on next login.

Do two things above and feedback to us for confirmation.

**Comment by mbnoimi** [comment]
>  Use kdesudo instead (Older KDE may use kdesu)

To get rid of any exception I installed a fresh Mint KDE 15 x64 on new machine then installed Customizer as usual (from install.sh as you mentioned above) but unfortunately it seems this issue not related to KDE at all it seems there is some thing wrong happen with Customizer itself. See the log please from here: http://pastebin.com/MyGMg97Y

Any way, I'll try to install Ubuntu 13.10 desktop x64 on another machine to see what of I can reproduce same bug or not.

**Comment by ghost** [comment]
@mbnoimi 
I have seen your logs posted on both Pastebin.com URL.

In the first log, I noticed that you mixed up `sudo` and `gksu` together in single command. This is incorrect.
Use `sudo` to run any CLI application from command line.
Use `gksu` (in GNOME, XFCE, etc.) or `kdesudo` (KDE variants) to run GUI application from command line.
But we don't use both command at the same time, ever.

In the second log, I can see you follow the command that I had adviced to you earlier. But the issue still happened. This was unexpected.

At the moment, I still suspect KDE host system or its component to be the cause.
One valid reason: This issue didn't happen on non-KDE host system.

Last weekend I did manage to set up a vanilla Linux Mint 15 (Cinnamon) 32-bit on my test machine. Despite the response was slow, I had successfully installed Customizer (gambas3) on it. And I even managed to load a randomly chosen ISO image (in this case, I used Ubuntu Mini Remix 12.04 32-bit).
No error was seen in Terminal output.

There are other possibilities to look for, but I will not discuss that here yet.

For now, I am expecting to hear your feedback.

**Comment by mbnoimi** [comment]
> Last weekend I did manage to set up a vanilla Linux Mint 15 (Cinnamon) 32-bit on my test machine. Despite the response was slow, I had successfully installed Customizer (gambas3) on it. And I even managed to load a randomly chosen ISO image (in this case, I used Ubuntu Mini Remix 12.04 32-bit).

When I installed Customizer under Ubuntu 13.10 x64 desktop things got worst as shown in the following log: http://pastebin.com/VbvK1jG4 (I got this error message when I called gksu /opt/Customizer/GUI.gambas as shown below)
![snapshot-9](https://f.cloud.github.com/assets/284547/1665126/3afd6c38-5c32-11e3-94ef-0807612922ec.png)

**Comment by mbnoimi** [comment]
After installing Gambas the error message disappeared and I successfully run Customize... Voila!

I suggest to add the missing packages in install.sh

P.S. I prefer to keep this ticket open because Customize doesn't work under KDE based distros.

**Comment by ghost** [comment]
@mbnoimi 
Now I feel much relieved after listening to your good news.

> I suggest to add the missing packages in install.sh

I will fix the install.sh script this weekend.
Can you tell us what are the missing packages?

> P.S. I prefer to keep this ticket open because Customize doesn't work under KDE based distros.

Yesterday, I tried to contact Customizer gambas3 port developer but seemed unreachable at the moment.

So far we do not receive much feedback from 64-bit users. As far as I know, you are the first 64-bit user that giving us continuous feedbacks. Thank you, we are really grateful for that!

On Customizer development team side, I have been testing only on 32-bit systems and mainly on LTS releases, due to lack of resources and time constraints. There's not much I can do, while working at the same time.

Until we can confirm this "Doesn't work under KDE" issue, I will mark this as "Pending".

Please feel free to give feedback to us from time to time.

**Comment by mbnoimi** [comment]
>  Can you tell us what are the missing packages?

I really don't know but I figured out there are missing packages from the above error message for that I installed whole gambas (I know this is a stupid solution but it fixed my issue)

> So far we do not receive much feedback from 64-bit users. As far as I know, you are the first 64-bit user that giving us continuous feedbacks. Thank you, we are really grateful for that!

Thanks a lot for intersting my feedback. Basically I'm using x64 so I'll start building my customized distro by Customizer soon and I wish to get a smooth test.

**Comment by ghost** [comment]
@mbnoimi 
I am planning to reproduce this issue, using 32-bit PC and Kubuntu 12.04 LTS 32-bit ISO. Although, will take some time to download ISO and prepare another test machine ( as my previous 8-year old test machine will be discarded soon ).

May be I will try end of this month, depends on how busy I would be for the next two weeks.
Just for your information.

**Comment by fluxer** [comment]
I've updated the install script and the issue with the gb.form.stock should be no more.

However, I think that the issue with the ISO not being loaded properly (widgets not being enabled) is related to bug in either Qt or Gambas3. Try to use more recent version of Gambas3 from third-party PPA and see if the issue still remains.

**Comment by mbnoimi** [comment]
@fluxer 

> I've updated the install script and the issue with the gb.form.stock should be no more.

Do you mean https://dl.dropboxusercontent.com/u/54183088/install.sh is no longer valid?

> Try to use more recent version of Gambas3 from third-party PPA and see if the issue still remains.

Do you mean it's better to use ppa:gambas-team/gambas-daily ?

**Comment by fluxer** [comment]
> Do you mean https://dl.dropboxusercontent.com/u/54183088/install.sh is no longer valid?

The link remains the same.

> Do you mean it's better to use ppa:gambas-team/gambas-daily ?

Yeah.

**Comment by ghost** [comment]
@fluxer 
I just noticed your updated script has copy-paste typo for gambas2-\* packages.

Line 224: `_GB_DEPENDS="gambas2-gb-gtk gambas2-gb-form gambas3-gb-form-stock`
`gambas2-gb-image gambas2-runtime gambas2-dev"`

Even we without the typo, the package `gambas2-gb-form-stock` is non-existent.

So, the fix should be delete the `gambas3-gb-form-stock` from Line 224.

**Comment by fluxer** [comment]
I will updated it in a minute.

Edit: Done!

**Comment by fluxer** [comment]
There are two reasons for Customizer (GUI) to disable the widgets after extracting the ISO - either /home/Filesystem/etc/casper.conf or /home/FileSystem/etc/lsb-release doesn't exist. I will fetch Linux Mint 15 KDE to check if those really do not exist.

**Comment by fluxer** [comment]
All good, I was even able to run nested X-session. Unless someone finds the source of your problem I'm considering this as invalid (there were two issues in this report, one of which solved, but not the main one).

**Comment by ghost** [comment]
@fluxer 
Noted.
I just managed to download Kubuntu 12.04 LTS 32-bit last night (continue from last week, took two days by part). But I don't have time to test now.. I have other things going on.

Kindly wait (before you mark as "invalid") until end of this month.
I will double-confirm the validity of this issue by then.

**Comment by fluxer** [comment]
@clearkimura Ok.

**Comment by ghost** [comment]
@fluxer 
Reproduced following test cases:

Case 1 -- Check KDE-based ISO

Host system:  Xubuntu 12.04 32-bit ( running in VirtualBox )
ISO source file:  Kubuntu 12.04 32-bit
-  Running Customizer:  OK
-  Extracting ISO:  OK
-  Using Desktop:  OK ( straight to Desktop, no fallback option )

Case 2 -- Check KDE-based host system

Host system:  Kubuntu 12.04 32-bit ( running in VirtualBox )
ISO source file:  Xubuntu 12.04 32-bit
-  Running Customizer:  OK ( using `kdesudo` command )
-  Extracting ISO:  FAILED

It seemed that this issue is valid, based on Case 2 result.

Something wrong when running Customizer under KDE-based host system.
Checking out other gambas3 packages..

**Comment by ghost** [comment]
@fluxer 
I tried to use daily PPA for gambas3 packages. Same issue occurs also--the ISO won't load at all and no Terminal window appears as it should.

>  QDBusConnection: session D-Bus connection created before QCoreApplication. Application may misbehave.

This error is common warning message that can be seen, even when running any KDE apps from Konsole. So never mind this.

>  QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
>  QFileSystemWatcher: failed to add paths: /root/.config/ibus/bus

But the latter are questionable but probably safe to ignore too. This error supposed to be solved by fix being patched to kde-workspace since 4:4.7.1-*, based on [this source](https://bugs.launchpad.net/ubuntu/+source/kde-workspace/+bug/855472). I tried installing `ibus` with its recommended packages, reboot, but that didn't eliminate these warning messages anyway.

On the other hand, CLI counterpart is not affected by this issue. So I attempt to extract via CLI, then tried running GUI. It runs with button widgets enabled. But even with this workaround, only "Edit sources" and "Install DEB" worked. Others remain unusable.

Perhaps KDE-based Terminal emulator is being incompatible? So I tried installing LXTerminal. And run Customizer GUI from it. No difference from using Konsole.

Above all, why Customizer GUI failed to load target ISO file, remains unknown at this point.

**Comment by fluxer** [comment]
Ok, it's probably an upstream issue - report the bug at the Gambas bug tracker. This is as far as I go on this one.

**Comment by fluxer** [comment]
No longer relevant, development has moved on.
